### PR TITLE
Flask: add logging to participants

### PR DESCRIPTION
**Pull request by Madelinehazel**
Related to #361

### CODE REVIEW

File: flask/app/participants.py [hunk 1]
```diff
@@ -39,23 +39,31 @@
...
             limit = int(limit)
             if limit <= 0:  # MySQL accepts 0 but that's just a waste of time
                 raise ValueError
     except:
+        app.logger.error("limit must be a positive integer")
         abort(400, description="limit must be a positive integer")
 
+    app.logger.debug("Parsing order query parameters")
+    app.logger.debug("Validating order_by")
     order_by = request.args.get("order_by", type=str)
```
Comment (delvinso) [inline flask/app/participants.py:67]: Could log the parsed 'order_by' argument to be in line with the rest of the query parameters?

File: flask/app/participants.py [hunk 2]
```diff
@@ -75,30 +83,38 @@ def list_participants() -> Response:
         # order_by has been restricted to a known list, so we can safely use getattr
         order = getattr(models.Participant, order_by)
     else:
+        app.logger.error("order_by must be one of %s", allowed_columns)
         abort(400, description=f"order_by must be one of {allowed_columns}")
 
+    app.logger.debug("Validating order_dir")
     if order:
         order_dir = request.args.get("order_dir", type=str)
```
Comment (delvinso) [inline flask/app/participants.py:91]: Same with 'order_dir'

File: flask/app/participants.py [hunk 3]
```diff
@@ -278,28 +320,38 @@ def update_participant(id: int):
...
-
+    app.logger.debug("Checking if participant already exists under family")
     ptp_query = models.Participant.query.filter(
         models.Participant.family_id == request.json.get("family_id"),
         models.Participant.participant_codename
         == request.json.get("participant_codename"),
     )
 
     if ptp_query.first() is not None:
+        app.logger.error("Participant codename already exists under family")
```
Comment (delvinso) [inline flask/app/participants.py:350]: I'm thinking from here to 377 values being queried could be logged for more granularity. something like `app.logger.error("Participant codename '%s' already exists under family ID", request.json.get("participant_codename")`.

**Review by delvinso — APPROVED** [review]
i might be guilty of not following this to a T for the logging I've done so far, but I was thinking for transparency and consistency any input/query parameters/etc. could be quoted. otherwise looks good to me

File: flask/app/participants.py [hunk 4]
```diff
@@ -75,30 +83,38 @@ def list_participants() -> Response:
...
             abort(400, description="order_dir must be either 'asc' or 'desc'")
 
+    app.logger.debug("Parsing filter query parameters")
     filters = []
     family_codename = request.args.get("family_codename", type=str)
+    app.logger.debug("Parsing family_codename: %s", family_codename)
     if family_codename:
         filters.append(func.instr(models.Family.family_codename, family_codename))
     participant_codename = request.args.get("participant_codename", type=str)
+    app.logger.debug("Parsing participant_codename: %s", participant_codename)
```
Comment (delvinso) [inline flask/app/participants.py:107]:
```suggestion
    app.logger.debug("Parsing participant_codename: '%s'", participant_codename)
```

File: flask/app/participants.py [hunk 5]
```diff
@@ -75,30 +83,38 @@ def list_participants() -> Response:
...
         elif order_dir == "asc":
             order = order.asc()
         else:
+            app.logger.error("order_dir must be one either 'asc', 'desc'")
             abort(400, description="order_dir must be either 'asc' or 'desc'")
 
+    app.logger.debug("Parsing filter query parameters")
     filters = []
     family_codename = request.args.get("family_codename", type=str)
+    app.logger.debug("Parsing family_codename: %s", family_codename)
```
Comment (delvinso) [inline flask/app/participants.py:103]:
```suggestion
    app.logger.debug("Parsing family_codename: '%s'", family_codename)
```

File: flask/app/participants.py [hunk 6]
```diff
@@ -75,30 +83,38 @@ def list_participants() -> Response:
         # order_by has been restricted to a known list, so we can safely use getattr
         order = getattr(models.Participant, order_by)
     else:
+        app.logger.error("order_by must be one of %s", allowed_columns)
         abort(400, description=f"order_by must be one of {allowed_columns}")
 
     if order:
         order_dir = request.args.get("order_dir", type=str)
+        app.logger.debug("Validating order_dir: %s", order_dir)
```
Comment (delvinso) [inline flask/app/participants.py:91]:
```suggestion
        app.logger.debug("Validating order_dir: '%s'", order_dir)
```

File: flask/app/participants.py [hunk 7]
```diff
@@ -75,30 +83,38 @@ def list_participants() -> Response:
         # order_by has been restricted to a known list, so we can safely use getattr
         order = getattr(models.Participant, order_by)
     else:
+        app.logger.error("order_by must be one of %s", allowed_columns)
```
Comment (delvinso) [inline flask/app/participants.py:86]:
```suggestion
        app.logger.error("order_by must be one of '%s'", allowed_columns)
```

File: flask/app/participants.py [hunk 8]
```diff
@@ -39,24 +39,32 @@
...
             limit = int(limit)
             if limit <= 0:  # MySQL accepts 0 but that's just a waste of time
                 raise ValueError
     except:
+        app.logger.error("limit must be a positive integer")
         abort(400, description="limit must be a positive integer")
 
+    app.logger.debug("Parsing order query parameters")
     order_by = request.args.get("order_by", type=str)
+    app.logger.debug("Validating order_by: %s", order_by)
```
Comment (delvinso) [inline flask/app/participants.py:67]:
```suggestion
    app.logger.debug("Validating order_by: '%s'", order_by)
```

File: flask/app/participants.py [hunk 9]
```diff
@@ -75,30 +83,38 @@ def list_participants() -> Response:
...
     if family_codename:
         filters.append(func.instr(models.Family.family_codename, family_codename))
     participant_codename = request.args.get("participant_codename", type=str)
+    app.logger.debug("Parsing participant_codename: %s", participant_codename)
     if participant_codename:
         filters.append(
             func.instr(models.Participant.participant_codename, participant_codename)
         )
     notes = request.args.get("notes", type=str)
+    app.logger.debug("Parsing notes: %s", notes)
```
Comment (delvinso) [inline flask/app/participants.py:113]:
```suggestion
    app.logger.debug("Parsing notes: '%s'", notes)
```

File: flask/app/participants.py [hunk 10]
```diff
@@ -75,30 +83,38 @@ def list_participants() -> Response:
...
     if notes:
         filters.append(func.instr(models.Participant.notes, notes))
     participant_type = request.args.get("participant_type", type=str)
+    app.logger.debug("Parsing participant_type: %s", participant_type)
```
Comment (delvinso) [inline flask/app/participants.py:117]:
```suggestion
    app.logger.debug("Parsing participant_type: '%s'", participant_type)
```

File: flask/app/participants.py [hunk 11]
```diff
@@ -108,22 +124,30 @@ def list_participants() -> Response:
             )
         )
     sex = request.args.get("sex", type=str)
+    app.logger.debug("Parsing sex: %s", sex)
```
Comment (delvinso) [inline flask/app/participants.py:127]:
```suggestion
    app.logger.debug("Parsing sex: '%s'", sex)
```

File: flask/app/participants.py [hunk 12]
```diff
@@ -108,22 +124,30 @@ def list_participants() -> Response:
...
     if sex:
         filters.append(filter_in_enum_or_abort(models.Participant.sex, models.Sex, sex))
     affected = request.args.get("affected", type=str)
+    app.logger.debug("Parsing affected: %s", affected)
```
Comment (delvinso) [inline flask/app/participants.py:131]:
```suggestion
    app.logger.debug("Parsing affected: '%s'", affected)
```

File: flask/app/participants.py [hunk 13]
```diff
@@ -108,22 +124,30 @@ def list_participants() -> Response:
...
     if affected:
         filters.append(
             filter_nullable_bool_or_abort(models.Participant.affected, affected)
         )
     solved = request.args.get("solved", type=str)
+    app.logger.debug("Parsing solved: %s", solved)
```
Comment (delvinso) [inline flask/app/participants.py:137]:
```suggestion
    app.logger.debug("Parsing solved: '%s'", solved)
```

File: flask/app/participants.py [hunk 14]
```diff
@@ -108,22 +124,30 @@ def list_participants() -> Response:
...
+    app.logger.debug("Retrieving user id")
     if app.config.get("LOGIN_DISABLED") or current_user.is_admin:
         user_id = request.args.get("user")
+        app.logger.debug("User is admin with ID '%s'", user_id)
     else:
         user_id = current_user.user_id
+        app.logger.debug("User is regular with ID '%s'", user_id)
+
     if user_id:  # Regular user or assumed identity, return only permitted participants
+        app.logger.debug("Processing query - restricted based on user id %s.", user_id)
```
Comment (delvinso) [inline flask/app/participants.py:150]:
```suggestion
        app.logger.debug("Processing query - restricted based on user id '%s'.", user_id)
```

File: flask/app/participants.py [hunk 15]
```diff
@@ -278,28 +320,44 @@ def update_participant(id: int):
...
+        app.logger.debug(
+            "LOGIN_DISABLED = True so updated and created by IDs are defaulting to 1"
+        )
         updated_by_id = 1
         created_by_id = 1
 
     # check if the participant exists under a given family
-
+    app.logger.debug(
+        "Checking if participant %s already exists under family",
```
Comment (delvinso) [inline flask/app/participants.py:343]:
```suggestion
        "Checking if participant '%s' already exists under family",
```

File: flask/app/participants.py [hunk 16]
```diff
@@ -278,28 +320,44 @@ def update_participant(id: int):
...
+    )
     ptp_query = models.Participant.query.filter(
         models.Participant.family_id == request.json.get("family_id"),
         models.Participant.participant_codename
         == request.json.get("participant_codename"),
     )
 
     if ptp_query.first() is not None:
+        app.logger.error(
+            "Participant codename %s already exists under family",
```
Comment (delvinso) [inline flask/app/participants.py:354]:
```suggestion
            "Participant codename '%s' already exists under family",
```

File: flask/app/participants.py [hunk 17]
```diff
@@ -278,28 +320,44 @@ def update_participant(id: int):
...
+            request.json.get("participant_codename"),
+        )
         abort(422, description="Participant codename already exists under family")
 
     # check if family exists
+    app.logger.debug("Checking if family %s exists", request.json.get("family_id"))
```
Comment (delvinso) [inline flask/app/participants.py:360]:
```suggestion
    app.logger.debug("Checking if family '%s' exists", request.json.get("family_id"))
```

File: flask/app/participants.py [hunk 18]
```diff
@@ -308,22 +366,27 @@ def create_participant():
     enum_error = enum_validate(models.Participant, request.json, editable_columns)
 
     if enum_error:
+        app.logger.error("%s", enum_error)
         abort(400, description=enum_error)
 
     # get institution id
+    app.logger.debug("Check if institution %s exists", request.json.get("institution"))
```
Comment (delvinso) [inline flask/app/participants.py:373]:
```suggestion
    app.logger.debug("Check if institution '%s' exists", request.json.get("institution"))
```

**Comment by kevinlul** [comment]
These are conflicting because of the progressive loading changes so I'm not sure how to proceed here.

**Pull request closed without merging**